src/briefcase/integrations/wix.py

```diff
@@ -154,14 +154,13 @@ def upgrade(self):
         """
         Upgrade an existing WiX install.
         """
-        if self.managed_install:
-            if self.exists():
-                self.command.logger.info("Removing old WiX install...")
-                self.command.shutil.rmtree(self.wix_home)
-
-                self.install()
-                self.command.logger.info("...done.")
-            else:
-                raise MissingToolError('WiX')
-        else:
+        if not self.managed_install:
             raise NonManagedToolError('WiX')
+        if self.exists():
+            self.command.logger.info("Removing old WiX install...")
+            self.command.shutil.rmtree(self.wix_home)
+
+            self.install()
+            self.command.logger.info("...done.")
+        else:
```

Comment:
Really minor - but given this PR has moved to an "if problem: return; do stuff" rather than "if ok: do stuff; else: return" pattern, this should probably be updated to do an "if not self.exists()" check.

Reply:
Updated!